aten/src/ATen/native/mkldnn/Linear.cpp

```diff
@@ -295,12 +295,15 @@ Tensor mkldnn_linear_pointwise_binary(
         input_reshaped.size(0), weight_t.size(0)};
     output = output.reshape(output_size_reshaped);
     other_reshaped = other_reshaped.reshape(output_size_reshaped);
+    TORCH_CHECK(
+        output.sizes() == other_reshaped.sizes(),
+        "linear_binary_run expects the size of output and other tensor to be the same");
+  } else {
+    TORCH_CHECK(
+        output.dim() == other_reshaped.dim(),
+        "linear_binary_run expects the dimension of output and other tensor to be the same");
```

Comment:
May I know why we check the `tensor.dim` in the second branch but `tensor.sizes` in the first branch?

Reply:
Because the inductor always reshapes the input of linear into a 2D tensor, it will always run into the second branch, so I only change the second branch to check the `tensor.dim` to support broadcast binary fusion.